scripts/docker-testnet/functions.sh

```diff
@@ -0,0 +1,196 @@
+#!/usr/bin/env bash
+
+# Starts from 3, if the DOCKER_NETWORK_SUBNET ends with a 0. The first IP address is reserved for the gateway and the
+# second one is allocated to the seednode. Therefore the counting starts from 3. If you modify the DOCKER_NETWORK_SUBNET
+# variable, make sure to change this one accordingly too.
+IP_HOST_BYTE=3
+
+cloneRepositories() {
+  if [[ -n $CI_RUN ]]; then
+    echo "Repositories have been cloned in the CI"
+  else
+    cd $(dirname $MULTIVERSXDIR)
+
+    git clone git@github.com:multiversx/mx-chain-deploy-go.git || true
+    git clone git@github.com:multiversx/mx-chain-proxy-go.git || true
+  fi
+}
+
+createDockerNetwork() {
+  docker network create -d bridge --subnet=${DOCKER_NETWORK_SUBNET} ${DOCKER_NETWORK_NAME}
+
+  # this variable is used to keep track of the allocated IP addresses in the network, by removing the last byte
+  # of the DOCKER_NETWORK_SUBNET. One can consider this the host network address without the last byte at the end.
+  export NETWORK_ADDRESS=$(echo "$DOCKER_NETWORK_SUBNET" | rev | cut -d. -f2- | rev)
+}
+
+startSeedNode() {
+  docker run -d --name seednode -v ${TESTNETDIR}/seednode/config:/go/mx-chain-go/cmd/seednode/config \
+  --network ${DOCKER_NETWORK_NAME} \
+  seednode:dev \
+  --rest-api-interface=0.0.0.0:10000
+}
+
+startObservers() {
+ local observerIdx=0
+ # Example for loop with injected variables in Bash
+ for ((i = 0; i < SHARDCOUNT; i++)); do
+   for ((j = 0; j < SHARD_OBSERVERCOUNT; j++)); do
+     # Your commands or code to be executed in each iteration
+     KEY_INDEX=$((TOTAL_NODECOUNT - observerIdx - 1))
+
+     docker run -d --name "observer${observerIdx}-${NETWORK_ADDRESS}.${IP_HOST_BYTE}-10200-shard${i}" \
+     -v $TESTNETDIR/node/config:/go/mx-chain-go/cmd/node/config \
+     --network ${DOCKER_NETWORK_NAME} \
+     node:dev \
+     --destination-shard-as-observer $i \
+     --rest-api-interface=0.0.0.0:10200 \
+     --config ./config/config_observer.toml \
+     --sk-index=${KEY_INDEX} \
+     $EXTRA_OBSERVERS_FLAGS
+
+
+     (( IP_HOST_BYTE++ ))
+     ((observerIdx++)) || true
+   done
+ done
+
+ for ((i = 0; i < META_OBSERVERCOUNT; i++)); do
+    KEY_INDEX=$((TOTAL_NODECOUNT - observerIdx - 1))
+
+    docker run -d --name "observer${observerIdx}-${NETWORK_ADDRESS}.${IP_HOST_BYTE}-10200-metachain" \
+        -v $TESTNETDIR/node/config:/go/mx-chain-go/cmd/node/config \
+        --network ${DOCKER_NETWORK_NAME} \
+        node:dev \
```

Comment:
please use same indentation pattern for the entire script and remove unnecessary empty lines

Reply:
I hope it's fixed.